### ISSUE TITLE
Wunsch: Verteilte Datenbanken (Schemata)

### ISSUE THREAD

**Issue by kroerig** [SVWS-NRW/SVWS-Server#170]
Bisher müssen alle Datenbanken (Schemata) auf deinem Server liegen. Das erzwingt die JSON Notation in der svwsconfig.json:

```
  "DBKonfiguration" : {
    "dbms" : "MARIA_DB",
    "location" : "mariadb",
    "defaultschema" : "svws",
    "SchemaKonfiguration" : [ {
      "name" : "svws",
      "svwslogin" : false,
      "username" : "...",
      "password" : "..."
    } ],
    "connectionRetries" : 0,
    "retryTimeout" : 0
  }
```

In Schild3 Desktop-Client aber ist es über die Connection-Files problemlos möglich Datenbanken auf unterschiedlichen Servern anzusprechen. Das sollte auch im SVWS Client möglich sein. In einer SaaS Umgebung wird man schon aus Lastgründen nicht nur einen DB-Server einsetzen.

Daher fände ich es besser, die _location_ pro _SchemaKonfiguration_ setzen zu können.

**Comment by FPfotenhauer** [comment]
Das Thema werden wir zu einem späteren Zeitpunkt angehen.
Ist aber auf unserer Agenda.

**Comment by T-Hagel-SK** [comment]
In einer SAAS-Umgebung wird es auch Sinn machen, mehr als einen SVWS-Server einzusetzen ...

Wäre es dann hier nicht sinnvoller, auf die Lösungen von Ribeka zu setzen ?

Dort startet der PC-Client zunächst gegen ein sog. Zentral-Schema, in dem wird dann für die jeweilge Schulnummer nachgeschaut, auf welche DB und welchen SVWS-Server die Schule zugreifen soll. Dafür gibt es das so. Schemaverwaltungs-Programm, mit dem man dann z.B. auch SVWS-Server wechseln könnte, falls einer überlastet oder down ist.

So zumindest die Theorie. Bislang habe ich allerdings noch keine Clients laufen, weil sich immer wieder Fehler finden. Wenn dann aber mal alles soweit rund läuft, sollte das eine feine Sache sein. Die PC-Clients könnten dann als ein Softwarepaket auf die PCs/Terminalserver verteilt werden und die Steuerung der Datenbanken und DB-, sowie SVWS-Server geschieht dann über dieses Schemaverwaltungsprogramm.

**Comment by kroerig** [comment]
Wenn ich das richtig verstanden haben, dann soll der SVWS Web-Client den klassischen (Windows gebundenen) PC-Client (SchILD) ablösen. Das wird zwar noch einige Zeit dauern, aber dann wäre Ribeka raus und der SVWS Server und Client müssten die Funktionalität selbst erfüllen können.

**Comment by T-Hagel-SK** [comment]
Ja, das habe ich auch so verstanden, aber erst für Schild4, wenn alles browserbasiert läuft. Bis dahin braucht es diesen Umweg über diese Schemaverwaltung.

Ob dann RIbeka raus ist, keine Ahnung. Wenn aber alles browserbasiert möglich ist, wird es trotzdem oder gerade deswegen Kompetenzen brauchen, um Anforderungen von Schulträgern, Schulämter, Schulverwaltungen, etc. browserbasiert bereitzustellen. Da könnte man durchaus auf die jahrelangen Erfahrungen von Ribeka zurückgreifen. Aber ob Ribeka oder wer auch immer, das ist dann nicht meine Baustelle :-)
